algebird-core/src/main/scala/com/twitter/algebird/BinaryClassificationAUC.scala

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2012 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.algebird
+
+/**
+ * Curve is a list of Confusion Matrices with different
+ * thresholds
+ *
+ * @param matrices List of Matrices
+ */
+case class Curve(matrices: List[ConfusionMatrix])
+
+/**
+ * Given a List of (x,y) this functions computes the
+ * Area Under the Curve
+ */
+object AreaUnderCurve {
+  private def trapezoid(points: Seq[(Double, Double)]): Double = {
+    require(points.length == 2)
+    val x = points.head
+    val y = points.last
+    (y._1 - x._1) * (y._2 + x._2) / 2.0
+  }
+
+  def of(curve: List[(Double, Double)]): Double = {
+    curve.toIterator.sliding(2).withPartial(false).aggregate(0.0)(
+      seqop = (auc: Double, points: Seq[(Double, Double)]) => auc + trapezoid(points),
+      combop = _ + _)
+  }
+}
+
+sealed trait AUCMetric
+case object ROC extends AUCMetric
+case object PR extends AUCMetric
+
+/**
+ * Sums Curves which are a series of Confusion Matrices
+ * with different thresholds
+ */
+case object CurveMonoid extends Monoid[Curve] {
+  def zero = Curve(Nil)
+  override def plus(left: Curve, right: Curve): Curve = {
```

Comment:
is there a sumOption implementation that would be more efficient that we should add here as well, for adding multiple curves at once?